In [5]:
import re

# Initialize the lists
mems = []
speeds = []
energies = []

other_dir = "../mem_speed_bench/"
with open(f'{other_dir}output_g_flickr_optimal_lo_long_acc.txt', 'r') as file:
    text = file.read()
    # Create blocks of text for each store activation
    blocks = text.split("==============================Store activation in INT2==============================")
    
    for block in blocks:
        # Find act mem
        mem_search = re.search('Act Mem: (\d+.\d+) MB', block)
        if mem_search is not None:
            mem = float(mem_search.group(1))
            mems.append(mem)
            
        # Find epoch/s
        speed_search = re.search('epoch/s: (\d+.\d+)', block)
        if speed_search is not None:
            speed = float(speed_search.group(1))
            speeds.append(speed)
            
        #Find energy - Get the second occurrence
        energy_search = re.findall('(\d+.\d+) kWh', block)
        if energy_search and len(energy_search) > 1:
            energy = float(energy_search[1])  # get the second energy occurrence
            energies.append(energy)

print("Mems: ", mems)
print("Speeds: ", speeds)
print("Energies: ", energies)

Mems:  []
Speeds:  []
Energies:  []


In [2]:
for i in range(0, len(mems), 5):
    if i == 0:
        print('mem avg:', sum(mems[i+2:i+5])/3)
        print('speed avg:', sum(speeds[i+2:i+5])/3)
        print('energy avg:', sum(energies[i+2:i+5])/3 * 1000)
        print()
    else:
        print('mem avg:', sum(mems[i:i+5])/5)
        print('speed avg:', sum(speeds[i:i+5])/5)
        print('energy avg:', sum(energies[i:i+5])/5 * 1000)
        print()

In [8]:
import re

highest_train = []
highest_train_stds = []
highest_valid = []
highest_valid_stds = []
final_train = []
final_train_stds = []
final_test = []
final_test_stds = []

with open(f'{other_dir}/output_g_flickr_no_optimal_lo_long_acc.txt', 'r') as file:
    lines = file.readlines()
    for index, line in enumerate(lines):
        if "All runs:" in line:
            for i in range(1, 5):  # Loop over the next 4 lines
                value_line = lines[index + i]
                value, std_dev = re.search(r'(\d+.\d+) ± (\d+.\d+)', value_line).groups()
                if 'Highest Train' in value_line:
                    highest_train.append(float(value))
                    highest_train_stds.append(float(std_dev))
                elif 'Highest Valid' in value_line:
                    highest_valid.append(float(value))
                    highest_valid_stds.append(float(std_dev))
                elif 'Final Train' in value_line:
                    final_train.append(float(value))
                    final_train_stds.append(float(std_dev))
                elif 'Final Test' in value_line:
                    final_test.append(float(value))
                    final_test_stds.append(float(std_dev))

print("Highest Train: ", sum(highest_train)/5)
print("Highest Train std: ", sum(highest_train_stds)/5)
print("Highest Valid: ", sum(highest_valid)/5)
print("Highest Valid std: ", sum(highest_valid_stds)/5)
print("Final Train: ", sum(final_train)/5)
print("Final Train std: ", sum(final_train_stds)/5)
print("Final Test: ", sum(final_test)/5)
print("Final Test std: ", sum(final_test_stds)/5)


Highest Train:  98.45
Highest Train std:  0.268
Highest Valid:  51.754
Highest Valid std:  0.158
Final Train:  60.334
Final Train std:  2.212
Final Test:  51.69799999999999
Final Test std:  0.20400000000000001


### no opt
- Highest Train:  98.45
- Highest Train std:  0.268
- Highest Valid:  51.754
- Highest Valid std:  0.158
- Final Train:  60.334
- Final Train std:  2.212
- Final Test:  51.69799999999999
- Final Test std:  0.20400000000000001

### with opt

- Highest Train:  98.56
- Highest Train std:  0.17400000000000002
- Highest Valid:  51.736000000000004
- Highest Valid std:  0.194
- Final Train:  59.544000000000004
- Final Train std:  2.056
- Final Test:  51.72800000000001
- Final Test std:  0.156